From 0ab30d06b3fad45da5fd209cfd216f5179091aab Mon Sep 17 00:00:00 2001
From: stasson <sebastien.tasson@intel.com>
Date: Tue, 20 Aug 2013 19:14:57 +0200
Subject: Dalvik:Optimization via CFLAG options

BZ: 102690

Add -fno-pic and -fno-pie to get faster execution

Category: device-enablement
Domain: AOSP-Dalvik-Runtime
Origin: internal
Upstream-Candidate: no, needs rework

Change-Id: Ib0ada6f3f533f143dda24f8d725fced0a13f7462
Orig-MCG-Change-Id: Ia1e92a13277f5fd4174121d8c47697daef451dd3
Signed-off-by: Sebastien MICHEL <sebastien.michel@intel.com>
Signed-off-by: stasson <sebastien.tasson@intel.com>
Reviewed-on: http://android.intel.com:8080/126541
Reviewed-by: Beyler, Jean Christophe <jean.christophe.beyler@intel.com>
Reviewed-by: lab_aqa <lab_aqa@intel.com>
Reviewed-by: Semukhina, Elena V <elena.v.semukhina@intel.com>
Reviewed-by: Chen, Dong-Yuan <dong-yuan.chen@intel.com>
Tested-by: Chen, Dong-Yuan <dong-yuan.chen@intel.com>
Reviewed-by: cactus <cactus@intel.com>
Tested-by: cactus <cactus@intel.com>
Signed-off-by: Serguei Katkov <serguei.i.katkov@intel.com>
---
 vm/Android.mk |    4 +++-
 1 files changed, 3 insertions(+), 1 deletions(-)

diff --git a/vm/Android.mk b/vm/Android.mk
index bd5aa38..4883821 100644
--- a/vm/Android.mk
+++ b/vm/Android.mk
@@ -45,7 +45,9 @@ include $(LOCAL_PATH)/ReconfigureDvm.mk
 
 ifeq ($(dvm_arch_variant),x86)
   $(info [TARGET] Building with x86 fast interpreter)
-  LOCAL_CFLAGS += -DMTERP_STUB
+  LOCAL_CFLAGS += -DMTERP_STUB \
+                  -fno-pic \
+                  -fno-pie
 endif
 ifeq ($(WITH_JIT),true)
   $(info [TARGET] Building with JIT)
-- 
1.7.4.1

